### PR TITLE
Fixes for relational triggers

### PR DESCRIPTION
**Pull request by ajreynol**
This makes several fixes for relational triggers, including:
- Proper orienting of triggers of the form x=a (matching on x), which sometimes were erroneously treated as a=x (matching on a),
- Limit equality matching to disequalities (x=y were erroneously treated as ~x=y). 
- Do not allow diseqaulity matching in combination with single constructor matching.
- Disable a heuristic (getTriggerWeight for relational triggers) that caused variable triggers in some cases to be treated at the same effort level as normal triggers,
- Fix an issue in isTermEligibleForInstantiation that accidentally allowed "instantiation constant terms" from CEGQI to mix with E-matching.

### CODE REVIEW

**Comment by ajreynol** [comment]
Results: +196-48 with respect to previous relational triggers on SMT-LIB, which is now +80-55 with respect to default.  I am tempted to make --relational-triggers on by default now.
```
                           Configuration  #unsat      time    #sat      time #solved    #total
                 CVC4-041819-2-fvet_relt   13203   10745.5       0         0     13203     23450
                    CVC4-041819-2-fvet_d   13178   10749.1       0         0     13178     23450
                        CVC4-041919_relt   13055     10454       0         0     13055     23450




For unsatisfiable : 
                    CVC4-041819-2-fvet_d       X      55     141
                 CVC4-041819-2-fvet_relt      80       X     196
                        CVC4-041919_relt      18      48       X
```

**Comment by ajreynol** [comment]
Updated:
```
                           Configuration  #unsat      time    #sat      time #solved    #total
                 CVC4-041819-2-fvet_relt   13203   10745.5       0         0     13203     23450
                    CVC4-041819-2-fvet_d   13178   10749.1       0         0     13178     23450
                        CVC4-041919_relt   13055     10454       0         0     13055     23450
                 CVC4-042219-2-fvet_relt   13206   10856.8       0         0     13206     23450





For unsatisfiable : 
                    CVC4-041819-2-fvet_d       X      55     141      56
                 CVC4-041819-2-fvet_relt      80       X     196       4
                        CVC4-041919_relt      18      48       X      46
                 CVC4-042219-2-fvet_relt      84       7     197       X
```

**Comment by HanielB** [comment]
Indeed, I think we will want to have this as default now.

File: test/regress/CMakeLists.txt [hunk 1]
```diff
@@ -1392,6 +1393,8 @@ set(regress_1_tests
   regress1/quantifiers/stream-x2014-09-18-unsat.smt2
   regress1/quantifiers/sygus-infer-nested.smt2
   regress1/quantifiers/symmetric_unsat_7.smt2
+  regress1/quantifiers/var-eq-trigger.smt2
+  regress1/quantifiers/var-eq-trigger-simple.smt2
```
Comment (HanielB) [inline test/regress/CMakeLists.txt]: add the regression of `regress2` as well

**Comment by ajreynol** [comment]
After latest fixes, gained another +6-0:

```
                           Configuration  #unsat      time    #sat      time #solved    #total
                 CVC4-041819-2-fvet_relt   13203   10745.5       0         0     13203     23450
                    CVC4-041819-2-fvet_d   13178   10749.1       0         0     13178     23450
                        CVC4-041919_relt   13055     10454       0         0     13055     23450
                 CVC4-042219-2-fvet_relt   13206   10856.8       0         0     13206     23450
                    CVC4-050819-2-fvet_d   13212   10720.8       0         0     13212     23450






For unsatisfiable : 
                    CVC4-041819-2-fvet_d       X      55     141      56      55
                 CVC4-041819-2-fvet_relt      80       X     196       4       3
                        CVC4-041919_relt      18      48       X      46      43
                 CVC4-042219-2-fvet_relt      84       7     197       X       0
                    CVC4-050819-2-fvet_d      89      12     200       6       X

```

**Pull request merged**